api/envoy/extensions/transport_sockets/tls/v3/common.proto

```diff
@@ -232,7 +233,49 @@ message TlsSessionTicketKeys {
       [(validate.rules).repeated = {min_items: 1}, (udpa.annotations.sensitive) = true];
 }
 
-// [#next-free-field: 13]
+// Config for Certificate provider to get certificates. This provider should allow certificates to be
+// fetched/refreshed over the network asynchronously with respect to the TLS handshake.
+// [#not-implemented-hide:]
+message CertificateProviderPlugin {
+  // opaque name used to specify certificate instances or types. For example, "ROOTCA" to specify
+  // a root-certificate (validation context) or "TLS" to specify a new tls-certificate.
+  string name = 1 [(validate.rules).string = {min_len: 1}];
```

Comment:
What is the difference of this `name` and the one in typed_config?

Reply:
The one in `typed_config` is simply a name or an opaque identifier for the extension. So it is purely config associated with the plugin. Whereas this name is meant to serve as an identification of the certificate that is meant to be provided by the certificate provider. Similar to the `certificate_name` on line 275. May be we should call this certificate_name too?

Reply:
After thinking about this for a bit, I've elected to just remove `CertificateProviderPlugin` completely.  We added that before we added `CertificateProviderPluginInstance`, but it's not actually being used anywhere, since all of our code uses the newer type.  We can always add this later if we actually need it for something.